### DISCUSSION TITLE
How to EDIT BAML?

### DISCUSSION THREAD

**Discussion by beastdrc** [dnSpyEx/dnSpy#132] [Q&A]
Hey! i want to fix some interface issues in some App, want to edit BAML\XAML 
Found a way to save XAML, but can't figure out how to edit and reexport?

**Comment by ElektroKill** [comment]
Hi, dnSpy does not currently allow the editing of XAML or its compiled version (BAML) which is embedded in a module. The only means of editing the BAML would be to obtain the binary representation of the BAML and replace the resource data with it. dnSpy does not have any features which make this easier as of now. I may consider adding a way to edit the BAML record structure in the future!

**Reply by beastdrc** [reply]
Is there any way to edit it, maybe using other tools? 
Or as I understand I could to modify it in c# code somwhere in program?

**Reply by ElektroKill** [reply]
The only way to edit XAML code right now is to fully decompile the application into a project file, apply the XAML and C# changes and recompile. Raw C# code editing can be performed from dnSpy when the code is not connected to any XAML/BAML.
